packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
+  /**
+   * The value of the option.
+   */
+  readonly value: string | number;
+  /**
+   * The label to appear beside the radio button.
+   */
+  readonly checked?: boolean;
+  readonly label?: string;
+  readonly onChange?: (valueIn: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export function SegmentedControlOption({
+  value,
+  checked,
+  children,
+  label,
+  onChange,
+}: SegmentedControlOptionProps) {
+  const { selectedOption, handleChange } = useSegmentedControl();
+  const localChecked = useMemo(() => {
+    return checked ?? selectedOption === value;
+  }, [selectedOption, checked]);
+  const inputId = `${value.toString()}_${useId()}`;
```

Comment:
Should be safe only needing useId

Reply:
We could, but I'm leaning toward keeping the `value` for better readability (`pizza_:r0:` vs. `:r0:`). I've had to dig into test outputs before that only referenced the `useId` dozens of times and it was _horrible_ to find what I wanted 😅  So for readability, debugging and uniqueness, I'd prefer to keep `value`.